In [1]:
import numpy as np
import pandas as pd

## Read JSON of pipeline data

In [2]:
df = pd.read_json('./info from WorkStep/candidate_funnel_2022_08.json')

In [5]:
df.shape

(40935, 15)

In [4]:
df['position_id'].nunique()

2670

There are 40k+ entries in the dataset, but only 2670 positions.

Let's look at a sample position which was hired for, to see what the pipeline looks like

In [19]:
df8049247818 = df[(df['company_id'] == 69615) & (df['position_id'] == 8049247818)].head(50)

In [20]:
df8049247818

company_id position_role_type  position_id                      date  \
36       69615          WAREHOUSE   8049247818 2022-08-01 00:00:00+00:00   
37       69615          WAREHOUSE   8049247818 2022-08-02 00:00:00+00:00   
38       69615          WAREHOUSE   8049247818 2022-08-03 00:00:00+00:00   
39       69615          WAREHOUSE   8049247818 2022-08-08 00:00:00+00:00   
40       69615          WAREHOUSE   8049247818 2022-08-11 00:00:00+00:00   
41       69615          WAREHOUSE   8049247818 2022-08-12 00:00:00+00:00   
42       69615          WAREHOUSE   8049247818 2022-08-14 00:00:00+00:00   
43       69615          WAREHOUSE   8049247818 2022-08-15 00:00:00+00:00   
44       69615          WAREHOUSE   8049247818 2022-08-16 00:00:00+00:00   
45       69615          WAREHOUSE   8049247818 2022-08-18 00:00:00+00:00   
46       69615          WAREHOUSE   8049247818 2022-08-21 00:00:00+00:00   
47       69615          WAREHOUSE   8049247818 2022-08-22 00:00:00+00:00   
48       69615          WAREHOUSE   8049247818 2022-08-29 00:00:00+00:00   

       spend  count_started  count_submitted  count_approved  count_reviewed  \
36  0.005527            2.0              1.0             1.0             NaN   
37  0.005857            NaN              NaN             NaN             1.0   
38  0.046157            1.0              1.0             1.0             NaN   
39  0.011225            2.0              1.0             1.0             NaN   
40  0.000549            1.0              1.0             1.0             3.0   
41  0.008016            1.0              NaN             NaN             NaN   
42  0.011640            1.0              2.0             2.0             NaN   
43  0.023024            3.0              NaN             NaN             NaN   
44  0.023866            1.0              1.0             1.0             2.0   
45  0.004319            1.0              1.0             1.0             NaN   
46  0.004905            1.0              NaN             NaN             NaN   
47  0.016032            1.0              1.0             NaN             2.0   
48       NaN            NaN              NaN             NaN             NaN   

    count_expired  count_withdrawn  count_deactivated  count_position_closed  \
36            NaN              NaN                NaN                    NaN   
37            NaN              NaN                NaN                    NaN   
38            NaN              NaN                NaN                    NaN   
39            NaN              NaN                NaN                    NaN   
40            NaN              NaN                NaN                    NaN   
41            NaN              NaN                NaN                    NaN   
42            NaN              NaN                NaN                    NaN   
43            NaN              NaN                NaN                    NaN   
44            NaN              NaN                NaN                    NaN   
45            NaN              NaN                NaN                    NaN   
46            NaN              NaN                NaN                    NaN   
47            NaN              NaN                NaN                    NaN   
48            NaN              NaN                NaN                    NaN   

    count_invited_to_interview  count_hired  
36                         NaN          NaN  
37                         1.0          NaN  
38                         NaN          NaN  
39                         NaN          NaN  
40                         3.0          NaN  
41                         NaN          NaN  
42                         NaN          NaN  
43                         NaN          NaN  
44                         1.0          NaN  
45                         NaN          NaN  
46                         NaN          NaN  
47                         1.0          NaN  
48                         NaN          1.0

In [24]:
summable_cols = ['spend', 'count_started', 'count_submitted', 'count_approved', 'count_reviewed',
       'count_expired', 'count_withdrawn', 'count_deactivated',
       'count_position_closed', 'count_invited_to_interview', 'count_hired']

In [25]:
df8049247818[summable_cols].sum()

spend                          0.161117
count_started                 15.000000
count_submitted                9.000000
count_approved                 8.000000
count_reviewed                 8.000000
count_expired                  0.000000
count_withdrawn                0.000000
count_deactivated              0.000000
count_position_closed          0.000000
count_invited_to_interview     6.000000
count_hired                    1.000000
dtype: float64

According to this data:

* 15 people began applications
* 9 completed them
* 8 passed initial screening
* 8 were reviewed by the HM
* 6 invited to interview
* 1 hired